### ISSUE TITLE
No longer build on Arch Linux

### ISSUE THREAD

**Issue by FabioLolix** [filebrowser/filebrowser#2532]
**Description**
Hello, I'm the maintainer of the pkgbuilds filebrowser and filebrowser-git on Arch Linux's AUR, Filebrowser no longer build from source on Arch Linux, both from git master and from stable releases; also previous stable release no longer builds

**Expected behaviour**
Build succede

**What is happening instead?**

Build fail

**Additional context**

I guess the problem is happening since Arch updated `openssl` from v1.1.x to v3.x

openssl v1.1.x exist and can be installed alongside but I have no idea how to pass that to the build system

Excerpt: 

```
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}
```

**How to reproduce?**
Install Arch Linux (or some close derivate)

```
sudo pacman -S --needed base-devel git
git clone https://aur.archlinux.org/packages/go.rice
cd go.rice
makepkg -si
cd ~
git clone https://aur.archlinux.org/packages/filebrowser-git
cd filebrowser-git
makepkg -si
```

go.rice is on AUR too and need to be build, for stable version git clone https://aur.archlinux.org/packages/filebrowser

**Files**
Build function log:
[filebrowser-git-2.23.0.r12.g6744cd47-1-x86_64-build.log](https://github.com/filebrowser/filebrowser/files/11952266/filebrowser-git-2.23.0.r12.g6744cd47-1-x86_64-build.log)

**Comment by FabioLolix** [comment]
Similar to https://github.com/filebrowser/filebrowser/issues/2545

**Comment by o1egl** [comment]
Currently frontend uses node 16. You can crate a PR to upgrade it to node 18

**Comment by FabioLolix** [comment]
> Currently frontend uses node 16

Arch still carries Node 16 https://archlinux.org/packages/extra/x86_64/nodejs-lts-gallium/ and is build against openssl v3.x, the build fails with the same error using that



> You can crate a PR to upgrade it to node 18

I don't have the knowledge for that sorry

**Comment by FabioLolix** [comment]
Bump

**Comment by adRn-s** [comment]
I'm using Arch Linux and I have installed extra/npm (version 10.1.0-1). And, I wasn't able to reproduce this issue:

```
filebrowser/frontend on  master via 🐹 v1.21.1 via  v20.7.0 
❯ npm install
npm WARN deprecated noty@3.2.0-beta-deprecated: no longer supported

added 460 packages, and audited 461 packages in 5s

69 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

filebrowser/frontend on  master via 🐹 v1.21.1 via  v20.7.0 took 5s 
❯ npm run build

> filebrowser-frontend@2.0.0 build
> vite build

vite v4.4.9 building for production...
<script src="[{[ .ReCaptchaHost ]}]/recaptcha/api.js?render=explicit"> in "/public/index.html" can't be bundled without type="module" attribute

[{[ .StaticURL ]}]/themes/[{[ .Theme ]}].css doesn't exist at build time, it will remain unchanged to be resolved at runtime

[{[ .StaticURL ]}]/custom.css doesn't exist at build time, it will remain unchanged to be resolved at runtime
✓ 183 modules transformed.
dist/assets/polyfills-legacy-5429e148.js.gz     55.56 kB
dist/assets/Editor-legacy-88a34ee5.js          876.75 kB │ gzip: 237.54 kB
dist/assets/index-legacy-6a5b3901.js         1,004.67 kB │ gzip: 309.72 kB

(!) Some chunks are larger than 500 kBs after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/configuration-options/#output-manualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
dist/assets/medium-vietnamese-20d9ffa9.woff2      4.78 kB
dist/assets/normal-vietnamese-0af602fe.woff2      4.90 kB
dist/public/index.html                            6.63 kB │ gzip:   2.19 kB
dist/assets/medium-greek-1320e5dd.woff2           7.18 kB
dist/assets/bold-vietnamese-2096d809.woff2        7.20 kB
dist/assets/normal-greek-adc0b6a1.woff2           7.21 kB
dist/assets/bold-greek-20b1d460.woff2             9.49 kB
dist/assets/normal-cyrillic-fb0297aa.woff2       10.00 kB
dist/assets/medium-cyrillic-ef372eb9.woff2       10.06 kB
dist/assets/bold-cyrillic-c72dc79d.woff2         11.68 kB
dist/assets/medium-latin-ext-b4dae108.woff2      11.91 kB
dist/assets/normal-latin-ext-55f25e8b.woff2      12.01 kB
dist/assets/normal-latin-f7bbc846.woff2          14.58 kB
dist/assets/medium-latin-01a44f86.woff2          14.60 kB
dist/assets/bold-latin-ext-2d83ee25.woff2        15.14 kB
dist/assets/medium-cyrillic-ext-2d290fe6.woff2   15.29 kB
dist/assets/normal-cyrillic-ext-457c1e0d.woff2   15.83 kB
dist/assets/bold-cyrillic-ext-d2a7562e.woff2     17.18 kB
dist/assets/bold-latin-827bd1a8.woff2            18.12 kB
dist/assets/material-icons-8265f647.woff2       128.35 kB
dist/assets/material-icons-fd84f88b.woff        164.91 kB
dist/assets/index-523be14b.css                   58.50 kB │ gzip:  16.78 kB
dist/assets/Editor-8ff5f267.js                  871.05 kB │ gzip: 242.89 kB
dist/assets/index-af40db8f.js                   927.56 kB │ gzip: 299.49 kB

(!) Some chunks are larger than 500 kBs after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/configuration-options/#output-manualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
✓ built in 41.46s
```

**Comment by adRn-s** [comment]
@FabioLolix can you provide a bit more of detail on what's your situation? I'd be happy to help

(disclaimer: I'm not related to filebrowser development)

**Comment by FabioLolix** [comment]
@adRn-s the problem have been resolved in time for `2.25.0` release, it build fine even on Node 20, you could see I have updated the stable pkgbuild to the version https://aur.archlinux.org/cgit/aur.git/commit/?h=filebrowser&id=34b6af9bed7ddd12abb8c5aaaba3348cd4807eb7

I'm just drowning on the pending work to do, I should have closed a week ago

> I have installed extra/npm (version 10.1.0-1)

the issues was node related; not npm related (Arch have 3 NodeJS versions to try) btw
